devclean.sh

```diff
@@ -61,115 +52,168 @@ usage_error () {
+REMOVE_CONDA=false
+REMOVE_SUB_MODULES=false
+CONTAINER=false
 
-# process optional arguments
+# process arguments
 while :; do
   case $1 in
     --help|-h) usage; exit 0 ;;
-    --all|-a) ALL_CLEAN=true ;;
-    --remove) REMOVE=true ;;
-    --remove=?*|--remove=) usage_error "$1 argument ignored." ;;
-    --clean) CLEAN=true ;;
-    --conda) REMOVE_CONDA=true ;;
-    --install-dir=?*) INSTALL_DIR=${1#*=} ;;
-    --install-dir|--install-dir=) usage_error "$1 requires argument." ;;
-    --build-dir=?*) BUILD_DIR=${1#*=} ;;
-    --build-dir|--build-dir=) usage_error "$1 requires argument." ;;
-    --bin-dir=?*) BIN_DIR=${1#*=} ;;
-    --bin-dir|--bin-dir=) usage_error "$1 requires argument." ;;
-    --conda-dir=?*) CONDA_DIR=${1#*=} ;;
-    --conda-dir|--conda-dir=) usage_error "$1 requires argument." ;;
-    --sub-modules) INCLUDE_SUB_MODULES=true ;;
+    --all|-a) REMOVE_BUILD=true; REMOVE_CONDA=true; REMOVE_SUB_MODULES=true ;;
+    --build|-b) REMOVE_BUILD=true ;;
+    --conda|-c) REMOVE_CONDA=true ;;
+    --container) CONTAINER=true ;;
+    --force) REMOVE=true ;;
+    --force=?*|--force=) usage_error "$1 argument ignored." ;;
+    --sub-modules|-s) REMOVE_SUB_MODULES=true ;;
+    --sub-modules=?*|--sub-modules=) usage_error "$1 argument ignored." ;;
     --verbose|-v) VERBOSE=true ;;
-    --verbose=?*|--verbose=) usage_error "$1 argument ignored." ;;
-    # targets
-    default) ALL_CLEAN=false ;;
     # unknown
     -?*|?*) usage_error "Unknown option $1" ;;
     *) break ;;
   esac
   shift
 done
 
-# choose defaults to clean
-if [ "${ALL_CLEAN}" = true ]; then
-  CLEAN=true
-fi
 
 # print settings
 if [ "${VERBOSE}" = true ] ; then
   settings
 fi
 
-# clean if build directory already exists 
-if [ "${REMOVE}" = true ] && [ "${CLEAN}" = false ] ; then
-  printf '%s\n' "Remove the \"build\" directory only, BUILD_DIR = $BUILD_DIR "
-  [[ -d ${BUILD_DIR} ]] && rm -rf ${BUILD_DIR} && printf '%s\n' "rm -rf ${BUILD_DIR}"
-elif [ "${CLEAN}" = true ]; then
-  printf '%s\n' "Remove build directory, bin directory, and other build artifacts "
-  printf '%s\n' " from the installation directory = ${INSTALL_DIR} "
-
-  directories=( \
-    "${BUILD_DIR}" \
-    "${INSTALL_DIR}/${BIN_DIR}" \
-    "${INSTALL_DIR}/share" \
-    "${INSTALL_DIR}/include" \
-    "${INSTALL_DIR}/lib" \
-    "${INSTALL_DIR}/lib64" \
+# Populate "removal_list" as an array of files/directories to remove, based on user selections
+declare -a removal_list='()'
+
+# Clean standard build artifacts
+if [ ${REMOVE_BUILD} == true ]; then
+  removal_list=( \
+    "${SRW_DIR}/build" \
+    "${SRW_DIR}/share" \
+    "${SRW_DIR}/include" \
+    "${SRW_DIR}/lib" \
+    "${SRW_DIR}/lib64" \
   )
-  if [ ${#directories[@]} -ge 1 ]; then
-    for dir in ${directories[@]}; do 
-     [[ -d "${dir}" ]] &&  rm -rfv ${dir} 
-    done
-  echo " "
-  fi
+  if [ ${CONTAINER} == true ]; then
+    removal_list+=("${SRW_DIR}/container-bin")
+  else
+    removal_list+=("${SRW_DIR}/exec")
 fi
```

Comment:
Requires a `fi` between lines 149 and 150.

```suggestion
  else
    removal_list+=("${SRW_DIR}/exec")
  fi
fi
```

Reply:
@MichaelLueken - thank you for noticing the old changes that creeped in. I restored devclean.sh to the commit 83f173c corresponding to PR #1100  https://github.com/ufs-community/ufs-srweather-app/pull/1100